samples/openTelemetryTracing.js

```diff
@@ -32,103 +32,111 @@
 
 const SUBSCRIBER_TIMEOUT = 10;
 
+// [START opentelemetry_tracing]
+/**
+ * TODO(developer): Uncomment these variables before running the sample.
+ */
+// const topicNameOrId = 'YOUR_TOPIC_OR_ID';
+// const subscriptionNameOrId = 'YOUR_SUBSCRIPTION_OR_ID';
+// const data = 'Hello, world!";
+
+// Imports the Google Cloud client library
+const {PubSub} = require('@google-cloud/pubsub');
+
+// Imports the OpenTelemetry API
+const otel = require('@opentelemetry/sdk-trace-node');
+const {diag, DiagConsoleLogger, DiagLogLevel} = require('@opentelemetry/api');
+const {NodeTracerProvider} = otel;
+const {SimpleSpanProcessor} = require('@opentelemetry/sdk-trace-base');
+
+// To output to the console for testing, use the ConsoleSpanExporter.
+const {ConsoleSpanExporter} = require('@opentelemetry/sdk-trace-base');
+
+// To output to Cloud Trace, import the OpenTelemetry bridge library.
+// import {TraceExporter} from '@google-cloud/opentelemetry-cloud-trace-exporter';
+
+const {Resource} = require('@opentelemetry/resources');
+const {
+  SemanticResourceAttributes,
+} = require('@opentelemetry/semantic-conventions');
+
+// Enable the diagnostic logger for OpenTelemetry
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG);
+
+// Log spans out to the console, for testing.
+const exporter = new ConsoleSpanExporter();
+
+// Log spans out to Cloud Trace, for production.
+// const exporter = new TraceExporter();
+
+// Build a tracer provider and a span processor to do
+// something with the spans we're generating.
+const provider = new NodeTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'otel example',
+  }),
+});
+const processor = new SimpleSpanProcessor(exporter);
+provider.addSpanProcessor(processor);
+provider.register();
+
+// Creates a client; cache this for further use.
+const pubSubClient = new PubSub();
+
+async function publishMessage(topicNameOrId, data) {
+  // Publishes the message as a string, e.g. "Hello, world!"
+  // or JSON.stringify(someObject)
+  const dataBuffer = Buffer.from(data);
+  const publisher = pubSubClient.topic(topicNameOrId);
+  const messageId = await publisher.publishMessage({data: dataBuffer});
+  console.log(`Message ${messageId} published.`);
+}
+
+async function subscriptionListen(subscriptionNameOrId) {
```

Comment:
I'm encountering a strange issue where (running this sample and using the cloud trace exporter), the `receive` span seems to get clobbered. Any ideas?

<img width="1193" alt="image" src="https://github.com/googleapis/nodejs-pubsub/assets/9397363/8069d6d4-9cca-4486-b8d7-79ceaa715ab1">

Reply:
This ended up being caused by me missing a bunch of places I needed to follow up on the renames earlier. I did some more renaming on stuff that had the ambiguous names, so hopefully that's cleaner all around.

<img width="1146" alt="image" src="https://github.com/googleapis/nodejs-pubsub/assets/57276408/6465c78c-a352-42f4-9636-af80706b5883">

Reply:
Yeah that fixed the receive span naming. Now I'm having an issue where the spans are still disconnected. Looking at the message directly, it looks like the otel propagation attributes are missing. Any idea what's wrong?

<img width="897" alt="image" src="https://github.com/googleapis/nodejs-pubsub/assets/9397363/429b0276-2fed-42e2-b2f0-424a0835da7d">

Reply:
Following up on this, I think there's something wrong I'm doing in my code since the basic sample includes the proper propagation.

Reply:
Ok I think I found the problem. Calling [`NodeTracerProvider.register()`](https://github.com/googleapis/nodejs-pubsub/blob/b218d5ff999eaf9e87c1cb4154f156679560c312/samples/typescript/openTelemetryTracing.ts#L75) makes this work as intended.

However, this isn't explicitly stated as required in the [Cloud Trace OpenTelemetry Node docs](https://cloud.google.com/trace/docs/setup/nodejs-ot), in [this sample](https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/blob/13997a0256e5124b5dda1c726db3f7c859e0a560/samples/trace/index.js#L48) specifically where it recommends calling `setGlobalTraceProvider(provider)` and does _not_ call `register`.

I'm not entirely sure what the difference is (since if you do both `register` and `setGlobalTraceProvider`), you get the error `Error: @opentelemetry/api: Attempted duplicate registration of API: trace` which seems to indicate these methods are doing the same thing. If `setGlobalTraceProvider` is not enough, we should file an issue to fix that sample.

Reply:
So using `setGlobalTraceProvider` isn't enough since `register` does additional things with the propagator. The Cloud Trace sample doesn't specify this since they don't use context propagation. As long as people are following our sample, it should work.